trestle/core/control_interface.py

```diff
@@ -0,0 +1,616 @@
+        for prop in as_list(imp_req.props):
+            if prop.name.startswith(const.RULE_NAME_PREFIX):
+                name = prop.value[len(const.RULE_NAME_PREFIX):]
+                remarks = string_from_root(prop.remarks)
+                rules[name] = remarks
+        return rules
+
+    @staticmethod
+    def get_params_from_imp_req(imp_req: comp.ImplementedRequirement) -> Dict[str, Dict[str, Any]]:
+        """Get all params found in this imp_req."""
+        params = {}
+        for param in as_list(imp_req.set_parameters):
+            values = [string_from_root(value) for value in param.values]
+            values = values[0] if len(values) == 1 else values
+            new_param = {'values': values}
+            if param.remarks:
+                new_param['remarks'] = string_from_root(param.remarks)
+            params[param.param_id] = new_param
+        return params
+
+    @staticmethod
+    def merge_dicts_deep(dest: Dict[Any, Any], src: Dict[Any, Any], overwrite_header_values: bool) -> None:
+        """
+        Merge dict src into dest.
+
+        New items are always added from src to dest.
+        Items present in both will be overriden dest if overwrite_header_values is True.
+        """
+        for key in src.keys():
+            if key in dest:
+                # if they are both dicts, recurse
+                if isinstance(dest[key], dict) and isinstance(src[key], dict):
+                    ControlInterface.merge_dicts_deep(dest[key], src[key], overwrite_header_values)
+                elif isinstance(dest[key], list) and isinstance(src[key], list):
+                    for item in src[key]:
+                        if item not in dest[key]:
+                            dest[key].append(item)
+                # otherwise override dest if needed
+                elif overwrite_header_values:
+                    dest[key] = src[key]
+            else:
+                # if the item was not already in dest, add it from src
+                dest[key] = src[key]
+
+    @staticmethod
+    def is_withdrawn(control: cat.Control) -> bool:
+        """
+        Determine if control is marked Withdrawn.
+
+        Args:
+            control: The control that may be marked withdrawn.
+
+        Returns:
+            True if marked withdrawn, false otherwise.
+
+        This is determined by property with name 'status' with value 'Withdrawn'.
+        """
+        for prop in as_list(control.props):
+            if prop.name and prop.value:
+                if prop.name.lower().strip() == 'status' and prop.value.lower().strip() == 'withdrawn':
+                    return True
+        return False
+
+    @staticmethod
+    def param_values_as_str_list(param: common.Parameter) -> List[str]:
+        """Convert param values to list of strings."""
+        return [val.__root__ for val in as_list(param.values)]
+
+    @staticmethod
+    def param_values_as_str(param: common.Parameter, brackets=False) -> Optional[str]:
```

Comment:
Shall those and many more methods be private instead to keep the API cleaner? I think only `param_to_str` should be public?

Reply:
I believe the intent of the single underscore on a method is to convey it is not to be used outside the class.  But these are all static methods intended to be used outside the class.  So I would leave them as "public" but we can change if desired.